homeassistant/components/mqtt/fan.py

```diff
@@ -212,7 +223,13 @@ async def async_setup_entry(
     config_entry: ConfigEntry,
     async_add_entities: AddEntitiesCallback,
 ) -> None:
-    """Set up MQTT fan dynamically through MQTT discovery."""
+    """Set up MQTT fan through configuration.yaml and dynamically through MQTT discovery."""
+    # load and initialize platform config from configuration.yaml
+    for config in await async_get_platform_config_from_yaml(
+        hass, fan.DOMAIN, PLATFORM_SCHEMA_MODERN
+    ):
+        await _async_setup_entity(hass, async_add_entities, config, config_entry)
```

Comment:
> We can probably gather these coroutines and await them concurrently.

You mean calling `hass.async_add_job()`?

Reply:
No, I think we can use `asyncio.gather` to gather and await all `_async_setup_entity` coroutines concurrently.

Reply:
Not sure we need to await here. @emontnemery what do you think?

Reply:
`_async_setup_entity` doesn't need to be a coroutine. We can change it to be a callback.

Reply:
We could, but would enlarge the scope of the PR.
We could do your suggestion though (since that part is new code):
```python
    await asyncio.gather(
        *(
            _async_setup_entity(hass, async_add_entities, config, config_entry)
            for config in await async_get_platform_config_from_yaml(
                hass, fan.DOMAIN, PLATFORM_SCHEMA_MODERN
            )
        )
    )
```